src/Analyzers/CSharp/Analyzers/UseCollectionExpression/UseCollectionExpressionHelpers.cs

```diff
@@ -613,118 +618,6 @@ bool IsPrimitiveConstant(ExpressionSyntax expression)
                semanticModel.GetTypeInfo(expression, cancellationToken).Type?.IsValueType == true;
     }
 
-    private static bool IsInTargetTypedLocation(SemanticModel semanticModel, ExpressionSyntax expression, CancellationToken cancellationToken)
```

Comment:
Moved this helper to common location.  It now returns the target-type for a particular location as the 'cast simplifier' now needs that info.